framework/behaviors/SluggableBehavior.php

```diff
@@ -171,6 +171,10 @@ protected function isNewSlugNeeded()
             return false;
         }
 
+        if ($this->attribute === null) {
```

Comment:
well, the current state of SluggableBehavior is too vague IMO, so every step to make it more readable is a step in the right direction.